libr/core/cmd_anal.c

```diff
@@ -2779,6 +2780,23 @@ static int cmd_anal_fcn(RCore *core, const char *input) {
 			r_core_cmdf (core, "\"afs %s\"", data);
 			free (sig);
 			free (data);
+		} else if (input[2] == 'r') { // "afsr"
+			if (input[3] == ' ') {
+				const char *ntype = ((input[4]) != '\0')
+					? (input + 4)
+					: NULL;
+				if (ntype) {
+					char *sig = r_core_cmd_str (core, "afs");
+					char *sig_notype = strchr (sig, ' ');
+					char *data = (char *)malloc (strlen (ntype) + strlen (sig_notype) + 1);
```

Comment:
just use r_str_newf. but ideally we shuold use the api not an r2 command

Reply:
> just use r_str_newf
Damn. This simplifies it a lot.

> but ideally we shuold use the api not an r2 command
We can leave it like this since it works, then refactor it. But, you know, nobody probably will do that in the near future. It could be refactored now, but it's gonna take some more time for me to deliver.